src/main/java/com/bootcamp/demo/restapi/controller/ScooterController.java

```diff
@@ -0,0 +1,51 @@
+package com.bootcamp.demo.restapi.controller;
+
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.restapi.assembler.ScooterAssembler;
+import com.bootcamp.demo.restapi.dto.scooter.ScooterRequestDTO;
+import com.bootcamp.demo.service.ScooterService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(path = "/api/scooters")
+public class ScooterController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterController.class);
+    private final ScooterService scooterService;
+
+    public ScooterController(ScooterService scooterService) {
+        this.scooterService = scooterService;
+    }
+
+    @GetMapping
+    @ResponseBody
+    public Iterable<Scooter> getScooters() {
+        LOGGER.info("GET SCOOTERS - API endpoint invoked");
+        return scooterService.findAllScooters();
+    }
+
+    @GetMapping("/{scooterId}")
+    public Scooter getScooterById(@PathVariable(value = "scooterId") String scooterId) {
+        LOGGER.info("GET SCOOTER BY ID - API endpoint invoked. scooterId = {}", scooterId);
+        return scooterService.findScooterById(scooterId);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Scooter postScooter(@Valid @RequestBody ScooterRequestDTO scooterRequestDTO) {
```

Comment:
Notes regarding the use of DTOs, **which should be discussed with the team**
- when using DTOs, they should be used both for input & output (since the idea is to provide another model, potentially simplified, to the client/user)
- discuss with colleagues about an approach for validation, since if the only reason for having DTOs is for validation (and are as result a wrapper over the model class), perhaps sth else might be more suitable